app/kubernetes/helpers/application/index.js

```diff
@@ -145,10 +145,12 @@ class KubernetesApplicationHelper {
   /* #endregion */
 
   /* #region  CONFIGURATIONS FV <> ENV & VOLUMES */
-  static generateConfigurationFormValuesFromEnvAndVolumes(env, volumes, configurations, filterCondition) {
+  static generateConfigurationFormValuesFromEnvAndVolumes(env, volumes, configurations, configurationKind) {
+    const filterCondition = configurationKind === KubernetesConfigurationKinds.CONFIGMAP ? 'valueFrom.configMapKeyRef.name' : 'valueFrom.secretKeyRef.name';
     const finalRes = _.flatMap(configurations, (cfg) => {
       const cfgEnv = _.filter(env, [filterCondition, cfg.Name]);
-      const cfgVol = _.filter(volumes, { configurationName: cfg.Name });
+      // const cfgVol = _.filter(volumes, { configurationName: cfg.Name });
```

Comment:
Can probably delete this commented out line.

Reply:
Nice spot! Updated